// node-appwrite/dist/enums/scopes.js
'use strict';

var Scopes = /* @__PURE__ */ ((Scopes2) => {
  Scopes2["SessionsWrite"] = "sessions.write";
  Scopes2["UsersRead"] = "users.read";
  Scopes2["UsersWrite"] = "users.write";
  Scopes2["TeamsRead"] = "teams.read";
  Scopes2["TeamsWrite"] = "teams.write";
  Scopes2["DatabasesRead"] = "databases.read";
  Scopes2["DatabasesWrite"] = "databases.write";
  Scopes2["CollectionsRead"] = "collections.read";
  Scopes2["CollectionsWrite"] = "collections.write";
  Scopes2["TablesRead"] = "tables.read";
  Scopes2["TablesWrite"] = "tables.write";
  Scopes2["AttributesRead"] = "attributes.read";
  Scopes2["AttributesWrite"] = "attributes.write";
  Scopes2["ColumnsRead"] = "columns.read";
  Scopes2["ColumnsWrite"] = "columns.write";
  Scopes2["IndexesRead"] = "indexes.read";
  Scopes2["IndexesWrite"] = "indexes.write";
  Scopes2["DocumentsRead"] = "documents.read";
  Scopes2["DocumentsWrite"] = "documents.write";
  Scopes2["RowsRead"] = "rows.read";
  Scopes2["RowsWrite"] = "rows.write";
  Scopes2["FilesRead"] = "files.read";
  Scopes2["FilesWrite"] = "files.write";
  Scopes2["BucketsRead"] = "buckets.read";
  Scopes2["BucketsWrite"] = "buckets.write";
  Scopes2["FunctionsRead"] = "functions.read";
  Scopes2["FunctionsWrite"] = "functions.write";
  Scopes2["SitesRead"] = "sites.read";
  Scopes2["SitesWrite"] = "sites.write";
  Scopes2["LogRead"] = "log.read";
  Scopes2["LogWrite"] = "log.write";
  Scopes2["ExecutionRead"] = "execution.read";
  Scopes2["ExecutionWrite"] = "execution.write";
  Scopes2["LocaleRead"] = "locale.read";
  Scopes2["AvatarsRead"] = "avatars.read";
  Scopes2["HealthRead"] = "health.read";
  Scopes2["ProvidersRead"] = "providers.read";
  Scopes2["ProvidersWrite"] = "providers.write";
  Scopes2["MessagesRead"] = "messages.read";
  Scopes2["MessagesWrite"] = "messages.write";
  Scopes2["TopicsRead"] = "topics.read";
  Scopes2["TopicsWrite"] = "topics.write";
  Scopes2["SubscribersRead"] = "subscribers.read";
  Scopes2["SubscribersWrite"] = "subscribers.write";
  Scopes2["TargetsRead"] = "targets.read";
  Scopes2["TargetsWrite"] = "targets.write";
  Scopes2["RulesRead"] = "rules.read";
  Scopes2["RulesWrite"] = "rules.write";
  Scopes2["SchedulesRead"] = "schedules.read";
  Scopes2["SchedulesWrite"] = "schedules.write";
  Scopes2["MigrationsRead"] = "migrations.read";
  Scopes2["MigrationsWrite"] = "migrations.write";
  Scopes2["VcsRead"] = "vcs.read";
  Scopes2["VcsWrite"] = "vcs.write";
  Scopes2["AssistantRead"] = "assistant.read";
  Scopes2["TokensRead"] = "tokens.read";
  Scopes2["TokensWrite"] = "tokens.write";
  Scopes2["WebhooksRead"] = "webhooks.read";
  Scopes2["WebhooksWrite"] = "webhooks.write";
  Scopes2["ProjectRead"] = "project.read";
  Scopes2["ProjectWrite"] = "project.write";
  Scopes2["PoliciesWrite"] = "policies.write";
  Scopes2["PoliciesRead"] = "policies.read";
  Scopes2["ArchivesRead"] = "archives.read";
  Scopes2["ArchivesWrite"] = "archives.write";
  Scopes2["RestorationsRead"] = "restorations.read";
  Scopes2["RestorationsWrite"] = "restorations.write";
  Scopes2["DomainsRead"] = "domains.read";
  Scopes2["DomainsWrite"] = "domains.write";
  Scopes2["EventsRead"] = "events.read";
  return Scopes2;
})(Scopes || {});

exports.Scopes = Scopes;
