// node-appwrite/dist/services/webhooks.js
'use strict';

var client = require('../client');

class Webhooks {
  constructor(client) {
    this.client = client;
  }
  list(paramsOrFirst, ...rest) {
    let params;
    if (!paramsOrFirst || paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        queries: paramsOrFirst,
        total: rest[0]
      };
    }
    const queries = params.queries;
    const total = params.total;
    const apiPath = "/webhooks";
    const payload = {};
    if (typeof queries !== "undefined") {
      payload["queries"] = queries;
    }
    if (typeof total !== "undefined") {
      payload["total"] = total;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  create(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst,
        url: rest[0],
        name: rest[1],
        events: rest[2],
        enabled: rest[3],
        security: rest[4],
        httpUser: rest[5],
        httpPass: rest[6]
      };
    }
    const webhookId = params.webhookId;
    const url = params.url;
    const name = params.name;
    const events = params.events;
    const enabled = params.enabled;
    const security = params.security;
    const httpUser = params.httpUser;
    const httpPass = params.httpPass;
    if (typeof webhookId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "webhookId"');
    }
    if (typeof url === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "url"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof events === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "events"');
    }
    const apiPath = "/webhooks";
    const payload = {};
    if (typeof webhookId !== "undefined") {
      payload["webhookId"] = webhookId;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof events !== "undefined") {
      payload["events"] = events;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof security !== "undefined") {
      payload["security"] = security;
    }
    if (typeof httpUser !== "undefined") {
      payload["httpUser"] = httpUser;
    }
    if (typeof httpPass !== "undefined") {
      payload["httpPass"] = httpPass;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "post",
      uri,
      apiHeaders,
      payload
    );
  }
  get(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst
      };
    }
    const webhookId = params.webhookId;
    if (typeof webhookId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", webhookId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {};
    return this.client.call(
      "get",
      uri,
      apiHeaders,
      payload
    );
  }
  update(paramsOrFirst, ...rest) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst,
        name: rest[0],
        url: rest[1],
        events: rest[2],
        enabled: rest[3],
        security: rest[4],
        httpUser: rest[5],
        httpPass: rest[6]
      };
    }
    const webhookId = params.webhookId;
    const name = params.name;
    const url = params.url;
    const events = params.events;
    const enabled = params.enabled;
    const security = params.security;
    const httpUser = params.httpUser;
    const httpPass = params.httpPass;
    if (typeof webhookId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "webhookId"');
    }
    if (typeof name === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "name"');
    }
    if (typeof url === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "url"');
    }
    if (typeof events === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "events"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", webhookId);
    const payload = {};
    if (typeof name !== "undefined") {
      payload["name"] = name;
    }
    if (typeof url !== "undefined") {
      payload["url"] = url;
    }
    if (typeof events !== "undefined") {
      payload["events"] = events;
    }
    if (typeof enabled !== "undefined") {
      payload["enabled"] = enabled;
    }
    if (typeof security !== "undefined") {
      payload["security"] = security;
    }
    if (typeof httpUser !== "undefined") {
      payload["httpUser"] = httpUser;
    }
    if (typeof httpPass !== "undefined") {
      payload["httpPass"] = httpPass;
    }
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "put",
      uri,
      apiHeaders,
      payload
    );
  }
  delete(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst
      };
    }
    const webhookId = params.webhookId;
    if (typeof webhookId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}".replace("{webhookId}", webhookId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "delete",
      uri,
      apiHeaders,
      payload
    );
  }
  updateSignature(paramsOrFirst) {
    let params;
    if (paramsOrFirst && typeof paramsOrFirst === "object" && !Array.isArray(paramsOrFirst)) {
      params = paramsOrFirst || {};
    } else {
      params = {
        webhookId: paramsOrFirst
      };
    }
    const webhookId = params.webhookId;
    if (typeof webhookId === "undefined") {
      throw new client.AppwriteException('Missing required parameter: "webhookId"');
    }
    const apiPath = "/webhooks/{webhookId}/signature".replace("{webhookId}", webhookId);
    const payload = {};
    const uri = new URL(this.client.config.endpoint + apiPath);
    const apiHeaders = {
      "content-type": "application/json"
    };
    return this.client.call(
      "patch",
      uri,
      apiHeaders,
      payload
    );
  }
}

exports.Webhooks = Webhooks;
